jax/_src/api_util.py

```diff
@@ -587,3 +583,79 @@ def _np_scalar_abstractify(x: np.generic) -> ShapedArray:
 # By default it does nothing, but it can be monkey-patched to do other things.
 def api_hook(fun, tag: str):
   return fun
+
+
+class TracingDebugInfo(NamedTuple):
+  traced_for: str             # e.g. 'jit', 'scan', etc
+  func_src_info: str          # e.g. f'{fun.__name__} at {filename}:{lineno}'
+  arg_names: Tuple[str, ...]  # e.g. ('args[0]', ... )
+
+def debug_info(traced_for: str, fun: Callable, args: Tuple[Any],
```

Comment:
Done. I didn't do it for `fun_sourceinfo` because that will become a module-internal function in the near future; I left a TODO to that effect instead.